gce/survival-training/dummy/train_test.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import json
+import os
+import random
+import tempfile
+import unittest
+
+from . import train
+
+
+class TrainTest(unittest.TestCase):
+
+  def setUp(self):
+    self.job_dir = tempfile.mkdtemp()
+    self.num_checkpoints = 10
+    self.checkpoint_files = []
+    self.checkpoint_steps = 100
+
+    self.test_job_dir = tempfile.mkdtemp()
+    self.test_job_file_glob = os.path.join(self.test_job_dir, "*")
+
+    # Note that hyperparameters are intended to be constant across checkpoints
+    self.hyperparameter_1 = 17
+    self.hyperparameter_2 = 3.14159
+
+    for i in range(self.num_checkpoints):
+      path = os.path.join(
+          self.job_dir,
+          "dummy-checkpoint-{}.json".format(i)
+      )
+
+      checkpoint_data = {
+          "steps": i*self.checkpoint_steps,
+          "hyperparameters": {
+              "hyperparameter_1": self.hyperparameter_1,
+              "hyperparameter_2": self.hyperparameter_2
+          },
+          "model": random.random()
+      }
+
+      with open(path, "w") as fp:
+        json.dump(checkpoint_data, fp)
+
+      self.checkpoint_files.append(path)
+
+    self.garbage_file = os.path.join(self.job_dir, "garbage")
+    with open(self.garbage_file, "w") as gf:
+      gf.write("garbage")
+
+  def tearDown(self):
+    os.remove(self.garbage_file)
```

Comment:
try/catch and anywhere else applicable.

Reply:
Even if I would just `raise`?

Reply:
Sorry are you asking if you need to try/catch if you simply raise?

Reply:
Yeah

Reply:
Yes even if you raise (or get an exception) you should catch the exception and do something meaningful with it. At minimum error log.